### PR TITLE
Return error if stripe POST method is called with GET

### PR DESCRIPTION
**Pull request by andrebola**
Users shouldn't get to this page with GET, but if they do now will
raise an error

### CODE REVIEW

**Comment by alastair** [comment]
It'd be good to see a test for this and the paypal view to check that this works properly

File: donations/views.py [hunk 1]
```diff
@@ -185,6 +185,8 @@ def donation_session_stripe(request):
               cancel_url=return_url_cancel,
             )
             return JsonResponse({"session_id":session.id})
+    # If request is GET return an error 400
```
Comment (alastair) [inline donations/views.py]: maybe update the comment - this will also happen if the form is invalid

**Comment by andrebola** [comment]
Added tests and also now I return form's errors in the same way that the paypal endpoint works.

File: donations/tests.py [hunk 2]
```diff
@@ -178,7 +179,57 @@ def test_annon_donation_stripe(self):
...
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
+            mock_create.return_value = session
+            data['amount'] = '0.1'
+            ret = self.client.post("/donations/donation-session-stripe/", data)
+            response =  ret.json()
+            # amount must be greater than 1
+            self.assertTrue('errors' in response)
+
+        with mock.patch('stripe.checkout.Session.create') as mock_create:
```
Comment (alastair) [inline donations/tests.py]: you could add a check to the tests that succeed to ensure that create() was called with the correct arguments. I understand that all of the ones that fail won't call this method, right?

File: donations/views.py [hunk 3]
```diff
@@ -185,6 +185,10 @@ def donation_session_stripe(request):
               cancel_url=return_url_cancel,
             )
             return JsonResponse({"session_id":session.id})
+        else:
+            return JsonResponse({'errors': form.errors})
```
Comment (alastair) [inline donations/views.py]: In the case that the form is invalid are you going to continue returning 200, or also 400?

File: donations/views.py [hunk 4]
```diff
@@ -185,6 +185,10 @@ def donation_session_stripe(request):
...
+    # If request is GET return an error 400
+    return HttpResponse(status=400)
```
Comment (alastair) [inline donations/views.py]: the paypal view returns a redirect to the homepage. Are we happy with the difference in behaviour?

**Comment by andrebola** [comment]
I think the difference is that the url for paypal is "by default" in the action of the form. So, in the case that someone has disabled js then he will be redirected. Does this make sense?

File: donations/views.py (hunk 3 above)
Comment (andrebola) [reply donations/views.py]: 200, like in 'donation_session_paypal'

File: donations/views.py (hunk 4 above)
Comment (andrebola) [reply donations/views.py]: see comment below

File: donations/tests.py (hunk 2 above)
Comment (andrebola) [reply donations/tests.py]: Done

**Comment by ffont** [comment]
Is this good to be merged now? @alastair @andrebola

**Comment by alastair** [comment]
yes, sure

**Pull request merged**